receiver/awscontainerinsightreceiver/internal/stores/utils.go

```diff
@@ -174,3 +178,92 @@ func refreshWithTimeout(parentContext context.Context, refresh func(), timeout t
 	<-ctx.Done()
 	cancel()
 }
+
+type RawContainerInsightsMetric struct {
+	// source of the metric for debugging merge conflict
+	ContainerName string
+	// key/value pairs that are typed and contain the metric (numerical) data
+	Fields map[string]any
+	// key/value string pairs that are used to identify the metrics
+	Tags   map[string]string
+	Logger *zap.Logger
+}
+
+var _ CIMetric = (*RawContainerInsightsMetric)(nil)
+
+func NewRawContainerInsightsMetric(mType string, logger *zap.Logger) *RawContainerInsightsMetric {
+	metric := &RawContainerInsightsMetric{
+		Fields: make(map[string]any),
+		Tags:   make(map[string]string),
+		Logger: logger,
+	}
+	metric.Tags[ci.MetricType] = mType
+	return metric
+}
+
+func NewRawContainerInsightsMetricWithData(mType string, fields map[string]any, tags map[string]string, logger *zap.Logger) *RawContainerInsightsMetric {
+	metric := NewRawContainerInsightsMetric(mType, logger)
+	metric.Fields = fields
+	metric.Tags = tags
```

Comment:
Is this meant to overwrite the `ci.MetricType` that was inserted into the `Tags` in the `NewRawContainerInsightsMetric` function? If not, then maybe you can just call the `AddTag`/`AddField` functions for the passed in maps.

Reply:
yes, type in tags parameter takes precedence.

Reply:
Then is there a point in passing the `mType` in? It's just going to get overwritten by the `tags`.

Reply:
I see your point. It is just to follow the existing convention of the initializer without data params.

Reply:
Okay I think it's just a bit misleading and a bit of unexpected behavior. Can we at least document?

```
// NewRawContainerInsightsMetricsWithData sets the fields and tags on the metric. Discards the mType.
```